### ISSUE TITLE
"withAmp" is not producing sole amp output on some pages

### ISSUE THREAD

**Issue by maddoxnelson** [zeit/next.js#7174]
# Bug report

## Describe the bug
I am using Next with an Express server, to take advantage of Express's routing and all of Next.js's fanciness. I am trying to create an app that has some Amp pages and some non-Amp pages (ideally, I would create a PWA AMP site, but I'm not quite there yet).

I've had AMP working fine on Next 7, but it required some not so pretty hacks to get working perfect. Today I upgraded to Next 8 and was very pleased to see how much code I could rip out of my project to get AMP working as expected with just the Next setup.

I have stumbled across one thing I think is a bug though.

According to the [Page Mode docs](https://nextjs.org/docs#amp-page-modes), I expect that by using `withAmp` I will get an "AMP-only" version of the page, that includes no extraneous JS, etc, and that it will be valid (or Next will alert me to validation errors).

However, what seems to happen is more like the behavior described in the "hybrid" section of those docs. I get a page that lets me know that Amp is available, but is not actually the AMP code I was expecting.

For example, this code:

```
import React, { Component } from 'react'
import { useAmp, withAmp } from 'next/amp'

class Sound extends Component {
    static async getInitialProps({ query }) {
        const { sound } = query;
        
        return {
          sound
        }
      }

    state = {
        sound: this.props.sound
    }

  render() {
    return (
      <div>
        {this.state.sound}
      </div>
    )
  }
}

export default withAmp(Sound)
```

Produces a page in which the Amp validator chrome extension turns _blue_ rather than green:

<img width="675" alt="Screen Shot 2019-04-26 at 1 46 37 PM" src="https://user-images.githubusercontent.com/4610199/56829584-c5068f00-6829-11e9-9842-d81ab90c4610.png">

I don't think this is the expected behavior, according to those docs? If I am misunderstanding, please let me know.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:
1. Pull down https://github.com/maddoxnelson/next-amp-bug (test repo I've made)
2. `npm install && npm run dev`
3. Visit `http://localhost:4375`
4. Notice that that page _isn't_ AMP -- that is the correct behavior.
5. Click "bark" or "woof" -- whichever is your favorite dog sound.
6. Or just go to `http://localhost:4375/dog-sounds/woof` which is my favorite dog sound.
7. If you have the chrome extension, note that it lets me know that AMP is available, but this is not actually AMP, just markup to direct me to an AMP version.
8. If you click through that chrome extension, you'll see it takes me to the same page but at`?amp=1`. I don't think this is really a bug since I'm on an express server and not handling that routing -- but in this case, I don't think I should have to?

**TL;dr I don't think that `withAmp()` is rendering only AMP markup, I think it's rendering hybrid markup.**

## System information
 - OS: [e.g. macOS, Windows]
 - Browser (if applies) Chrome
 - Version of Next.js: Canary -- I think 8.1.16-canary (but it is specified in the test repo)

*Additional Context:*
I have a sneaking suspicion that this might have something to do with the AMP Optimizer, but maybe I just don't understand enough about how that works.

**Comment by maddoxnelson** [comment]
Okay update: I fixed it, but I'm not sure if I've done a "best practice" where all this is concerned.

Leaving a note in case it's helpful!

Basically, my express app was stripping out the query parameters, so I was able to make sure they were back in by doing this: https://github.com/maddoxnelson/next-amp-bug/blob/master/server.js#L20-L22

I don't know if that's a _good_ thing... but it does work.

I'm going to leave this open, in case someone sees this and sees a better way (or maybe feels like dropping some knowledge on me)? But feel free to close.

Hopefully this helps someone else out in the future.

Loving the AMP support so far, btw!

**Comment by ijjk** [comment]
Hi, your suspicion that it has to do with the AMP Optimizer is correct. For AMP only pages we create two versions a dirty optimized and a clean optimized. To view the clean optimized version of an AMP only page you add `?amp=1` to your URL. 

Note: in development, we add script tags to provide automatic reloading of the page.

**Comment by maddoxnelson** [comment]
Thank you for the speedy reply @ijjk, and thank you for taking the time to explain it to me!

I'll close this now. I think there's enough information to unstick anyone who run into the same thing.

**Comment by k-prashant** [comment]
> To view the clean optimized version of an AMP only page you add `?amp=1` to your URL.

Can we use any alternate way to view amp pages like adding `/amp` or include  `.amp` in url?

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
